### ISSUE TITLE
Training with validation data

### ISSUE THREAD

**Issue by theoVag** [voxelmorph/voxelmorph#159]
Hello,
I have trained the model for my dataset and I thought that adding validation data in the training phase can help me avoid overfit and improve results. The sublosses for validation, like spatial val_transformer,loss,val_flow_loss are reasonable but the val_loss is not the weighted sum of the previous. Could you tell me if you have tried training with validation data and what changes are required?

Thanks a lot!!!

**Comment by adalca** [comment]
@theoVag 

- what do you mean 'training with validation' -- do you mean you use model.fit_generator and provide a validation generator? Or do you mean adding more data to the training?

- what makes you think you are overfitting?

- for your specific question of adding losses, can you please provide the output specifically? As much organized information as you can (input, output, exactly what you did, etc) would be great.

**Comment by theoVag** [comment]
Hello, I have just found the solution. The problem solved when I downgraded the keras to 2.2.5
